cmake/modules/extensions.cmake

```diff
@@ -1571,8 +1577,13 @@ endfunction()
 # `alpha_soc_bar_1_0_0;alpha_soc_bar` in `build_string` parameter.
 # `alpha_bar_1_0_0;alpha_bar` in `short_build_string` parameter.
 #
+# calling
+#   zephyr_build_string(build_string SOC BOARD_QUALIFIERS /soc/bar/foo MERGE)
+# will return a list of the following strings
+# `soc_bar_foo;soc` in `build_string` parameter.
```

Comment:
Not sure we should support MERGE from the start, based on experiences from #70548 and #71149 

In general we cannot expect that all overlays can apply cleanly, just think of the `rp2040.overlay` being applied for a `<board>/rp2040/w` build.

Let's re-consider a MERGE feature for qualifiers when addressing https://github.com/zephyrproject-rtos/zephyr/issues/69548.

Reply:
Removed, did not have rp2040 in mind when testing